itwin-platform-access/imodels-access-backend/src/BackendIModelsAccess.ts

```diff
@@ -254,6 +254,45 @@ export class BackendIModelsAccess implements BackendHubAccess {
     return rangeTotalBytes;
   }
 
+  // The imodels api does not distinguish between a v2 and a v1 checkpoint when calling getCurrentOrPrecedingCheckpoint.
+  // It is possible that a preceding v2 checkpoint exists, but earlier in the timeline than the most recent v1 checkpoint. In this case we would miss out on the preceding v2 checkpoint. 
+  // To get around this, this function decrements the changesetIndex of the discovered checkpoint if it is not a v2 checkpoint and searches again. 
+  private async queryCurrentOrPrecedingV2CheckpointHelper(arg: CheckpointProps, changesetIndex: number): Promise<ContainerAccessInfo | undefined> {
```

Comment:
To avoid the `helper` here maybe we could name this `findLatestV2CheckpointForChangeset`?

Reply:
good call

Reply:
opened new pR here: https://github.com/iTwin/imodels-clients/pull/149/files